### PR TITLE
MessageBus: To increase concurrency for a message_type

### PR DESCRIPTION
**Pull request by selvakumaar5496**
Signed-off-by: Selvakumar <selva.k.nambi@seagate.com>

### CODE REVIEW

File: py-utils/src/utils/message_bus/message_broker_collection.py [hunk 1]
```diff
@@ -191,39 +191,38 @@ def deregister_message_type(self, admin_id: str, message_types: list):
...
+        concurrency_count   Integer that represents number of partitions to be
+                            increased.
+
+        Note:  Number of partitions for a message type can only be increased,
+               never decreased
         """
         admin = self._clients['admin'][admin_id]
-        new_partition = [NewPartitions(each_message_type, \
-            new_total_count=partitions) for each_message_type in message_types]
+        new_partition = [NewPartitions(message_type, \
```
Comment (sachinpunadikar) [inline py-utils/src/utils/message_bus/message_broker_collection.py:210]: Add check for if the new value is less than or equal to existing value.
If so, do nothing and return error / success likewise. Error should say "can not decrease the concurrency count"

File: py-utils/src/utils/message_bus/message_broker_collection.py [hunk 2]
```diff
@@ -191,39 +191,38 @@ def deregister_message_type(self, admin_id: str, message_types: list):
...
-                    [each_message_type].__dict__['partitions']):
-                    if list_retry > self._max_list_message_type_count:
-                        raise MessageBusError(errno.EINVAL, "Maximum retries \
-                            exceeded to increase partition for %s.", \
-                            each_message_type)
-                    time.sleep(list_retry*1)
-                    continue
-                else:
-                    break
+        for list_retry in range(1, self._max_list_message_type_count+2):
```
Comment (sachinpunadikar) [inline py-utils/src/utils/message_bus/message_broker_collection.py:215]: Add a comment on what is this retry for ?

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
Work on the comments

File: py-utils/src/utils/message_bus/message_broker_collection.py (hunk 1 above)
Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_broker_collection.py:210]: Kafka handles this error

File: py-utils/src/utils/message_bus/message_broker_collection.py (hunk 2 above)
Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_broker_collection.py:215]: done. added

File: py-utils/src/utils/message_bus/message_broker_collection.py [hunk 3]
```diff
@@ -191,39 +191,39 @@ def deregister_message_type(self, admin_id: str, message_types: list):
...
-                    [each_message_type].__dict__['partitions']):
-                    if list_retry > self._max_list_message_type_count:
-                        raise MessageBusError(errno.EINVAL, "Maximum retries \
-                            exceeded to increase partition for %s.", \
-                            each_message_type)
-                    time.sleep(list_retry*1)
-                    continue
-                else:
-                    break
+        # Retries for an amount of time to increase the partition
```
Comment (sachinpunadikar) [inline py-utils/src/utils/message_bus/message_broker_collection.py:215]: Better wording would be
"Waiting for few seconds/milliseconds to complete the partition addition process"
Mention exact waiting period if known

**Review by sachinpunadikar — APPROVED** [review]
+1
A minor comment. Work on that

File: py-utils/src/utils/message_bus/message_broker_collection.py [hunk 4]
```diff
@@ -191,39 +191,39 @@ def deregister_message_type(self, admin_id: str, message_types: list):
                 else:
                     break
 
-    def increase_parallelism(self, admin_id: str, message_types: list, \
-        partitions: int):
+    def add_concurrency(self, admin_id: str, message_type: str, \
```
Comment (ujjwalpl) [inline py-utils/src/utils/message_bus/message_broker_collection.py:194]: Concurrency and replication are two things. Concurrency is the performance measure. This one is more related to replication.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/message_bus/message_broker_collection.py (hunk 4 above)
Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_broker_collection.py:194]: Discussed and finalized to go with concurrency

**Review by ujjwalpl — APPROVED** [review]

**Comment by mukul-seagate11** [comment]
ok to test

**Comment by mukul-seagate11** [comment]
run mini-provisioning

**Pull request merged**